services/inc/logging.h

```diff
@@ -433,7 +431,7 @@ static const char* const _log_category = NULL;
 #define PANIC(_code, _fmt, ...) \
         do { \
             LOG_DEBUG(PANIC, _fmt, ##__VA_ARGS__); \
-            panic_(_code, NULL, HAL_Delay_Microseconds); \
```

Comment:
A second thought/result of private discussion: this needs to be kept as-is for Photon/P1/Electron. The problem is that services dynalib exists in system-part1 and timer_hal implementation in part2. This is the opposite way of the dependency tree, the imports go from part1 into part2 into user app, hence why the pointer to micros function is passed here. Fine on other platforms, but going cause issues for Gen 2.